### ISSUE TITLE
Latest Docker: RAW conversion with darktable fails with 'Operation not permitted'

### ISSUE THREAD

**Issue by sir-Gollum** [photoprism/photoprism#1449]
Hi, thanks a lot for the app!

I started loading my photo library into it and encountered an issue I wasn't able to find here. I have a number of `.CR2` RAW files. If my understanding is correct, Photoprism tries to convert them to `.jpg` during indexing and fails with the following error:

```
index: failed converting 6574.CR2 to jpeg (path lookup '/usr/bin/../share/darktable' fails with: 'Operation not permitted' )
```

I was able to reproduce it on Mac by running
```
$ docker run --rm  photoprism/photoprism:latest /entrypoint.sh darktable-cli
running as uid 0
/entrypoint.sh darktable-cli
running as uid 0
darktable-cli
path lookup '/usr/bin/../share/darktable' fails with: 'Operation not permitted'
```
and on Linux by `docker exec`-ing into a running container and running `darktable-cli`. 

In an attempt to fix it, i tried to reinstall darktable, but `apt-get update` failed a number of GPG checks errors such as the following:
```
W: GPG error: http://archive.ubuntu.com/ubuntu hirsute InRelease: gpgv, gpgv2 or gpgv1 required for verification, but neither seems installed
E: The repository 'http://archive.ubuntu.com/ubuntu hirsute InRelease' is not signed.
```

I'm using the latest docker image for Photoprism (digest `40e99d97615164f1a10c2a674301ced878030a7a79df1b8b40d70f096c4112a9`), any help would be appreciated. Thanks again!

**Comment by sir-Gollum** [comment]
Tried installing darktable into `ubuntu:21.04` image, which the latest Photoprism images is based on, and `darktable-cli` seems to work - at least it shows a help message and doesn't throw an error when called without parameters.

**Comment by lastzero** [comment]
Did you try our development preview as well? Surprised Darktable is missing and nobody noticed this so far. Note you can disable it and use Rawtherapee instead. If Darktable isn't installed, it should be skipped. This is very strange.

**Comment by sir-Gollum** [comment]
Yes, I tried disabling Darktable, but it also seems broken in the latest docker:
```
index: failed converting 6565.CR2 to jpeg (terminate called after throwing an instance of 'Glib::ConvertError' )
```

Tried `photoprism/photoprism:preview` image just now, with success:
```
$ docker run --rm -it photoprism/photoprism:preview /entrypoint.sh darktable-cli
running as uid 0
/entrypoint.sh darktable-cli
running as uid 0
darktable-cli
usage: darktable-cli [<input file or dir>] [<xmp file>] <output destination> [options] [--core <darktable options>]

options:
   --width <max width> default: 0 = full resolution
   --height <max height> default: 0 = full resolution
   --bpp <bpp>, unsupported
   --hq <0|1|false|true> default: true
   --upscale <0|1|false|true>, default: false
   --export_masks <0|1|false|true>, default: false
   --style <style name>
   --style-overwrite
   --apply-custom-presets <0|1|false|true>, default: true
                          disable for multiple instances
   --out-ext <extension>, default from output destination or '.jpg'
                          if specified, takes preference over output
   --import <file or dir> specify input file or dir, can be used'
                          multiple times instead of input file
   --icc-type <type> specify icc type, default to NONE
                     use --help icc-type for list of supported types
   --icc-file <file> specify icc filename, default to NONE
   --icc-intent <intent> specify icc intent, default to LAST
                     use --help icc-intent for list of supported intents
   --verbose
   --help,-h [option]
   --version
```
I guess I'll try switching to that for some time, thanks!

**Comment by lastzero** [comment]
Any reason you used the entry point script as a wrapper instead of running Darktable directly? That may make a difference.

**Comment by sir-Gollum** [comment]
The successful run above was from a Mac where I tested. However, when I updated a Linux machine to use `photoprism/photoprism:preview`, I still get the `Operation not permitted` error running the same command (both with and without entrypoint). 

Re: entrypoint, no particular reason. I was just trying to figure out what's going on and inspected the container, plus I found that entrypoint is being set up in the [Dockerfile](https://github.com/photoprism/photoprism/blob/develop/docker/photoprism/Dockerfile#L140-L144). Running `darktable-cli` also fails under `root` in the container.

**Comment by lastzero** [comment]
Might be an issue with Unraid? Our demo uses the development preview and runs on Linux. A CR2 RAW could be uploaded and converted:

![Screenshot_20210727_105752](https://user-images.githubusercontent.com/301686/127126854-fa2c93dc-4438-4e34-8892-8092b9c0b0b0.jpg)

**Comment by lastzero** [comment]
Did you enable any security features / Linux kernel security?

**Comment by lastzero** [comment]
Note we're on vacation this week and out of office.

**Comment by sir-Gollum** [comment]
After some more digging, I found a solution for my issue in https://stackoverflow.com/questions/66319610/gpg-error-in-ubuntu-21-04-after-second-apt-get-update-during-docker-build, posting here for posterity.

It seems like the darktable issue and the `apt-get update` issue are related to a bug in a combination of docker / glibc. I wasn't seeing the issue on Mac because I have a newer Docker version (20.10.7) there, while the Unraid machine had Docker 19.03.14. 
Per suggestion in the StackOverflow thread, I updated Unraid to the latest version available (6.9.1 -> 6.9.2) and that brought me Docker 20.10.5, where the issue is gone.

Thanks for your quick and helpful responses Lastzero, and enjoy your vacation!
